pip/commands/configuration.py

```diff
@@ -0,0 +1,242 @@
+        )
+
+        self.cmd_opts.add_option(
+            '--venv',
+            dest='venv_file',
+            action='store_true',
+            default=False,
+            help='Use the virtualenv configuration file only'
+        )
+
+        self.parser.insert_option_group(0, self.cmd_opts)
+
+    def run(self, options, args):
+
+        # Determine what action is to be taken
+        action_options = {
+            "list": options.list,
+            "edit": options.edit,
+            "get": options.get_name,
+            "set": options.set_name_value,
+            "unset": options.unset_name,
+        }
+
+        action = None
+        for k, v in action_options.items():
+            if v:
+                if action is not None:
+                    # this works because of the conditional immediately after
+                    # the loop.
+                    action = None
+                    break
+                action = k
+
+        if action is None:
+            logger.error(
+                "Need exactly one action (--list, --edit, "
+                "--get, --set, --unset) to perform."
+            )
+            self.parser.print_help()
+            return ERROR
+
+        # Determine which configuration files are to be loaded
+        if sum([options.user_file, options.global_file, options.venv_file]):
+            logger.error(
+                "Need at-most one configuration file to use - pass "
+                "only one of --global, --user, --venv."
+            )
+            return ERROR
+
+        kwargs = {}
+        if options.user_file:
+            kwargs["load_only"] = "user"
+        elif options.global_file:
+            kwargs["load_only"] = "site-wide"
+        elif options.venv_file:
+            kwargs["load_only"] = "venv"
+        elif action in ["set", "unset", "edit"]:
+            logger.error(
+                "Need one configuration file to modify - pass one of "
+                "--global, --user, --venv."
+            )
+            return ERROR
+
+        # Load a new configuration
+        self.configuration = Configuration(
+            isolated=options.isolated_mode, **kwargs
+        )
+        self.configuration.load()
+
+        # Call the handler for the action with the options
+        handlers = {
+            "list": self.list_values,
+            "edit": self.open_in_editor,
+            "get": self.get_name,
+            "set": self.set_name_value,
+            "unset": self.unset_name
+        }
+
+        return handlers[action](options)
+
+    def list_values(self, options):
+        for key, value in sorted(self.configuration.items()):
+            logger.info("%s=%r", key, value)
+        return SUCCESS
+
+    def open_in_editor(self, options):
+        if options.editor is None:
+            logger.error(
+                "--edit requires an editor to be passed, either using "
+                "--editor or by setting it in a configuration file."
```

Comment:
I don't see an ``--editor`` flag defined up above, did I miss it? It would also be nice to support the semi standard ``$EDITOR`` environment variable.